Assets/Scripts/WinOrLoseScript.cs

```diff
@@ -54,88 +164,266 @@ void Start()
                     break;
             }
         }
+    }
 
-        if(players.Length != 4)
+    void MakeLastPlayerCry()
+    {
+        if (!SinglePlayer)
         {
-            players[players.Length - 1].GetComponent<Animator>().SetBool("Fourth", true);
+            List<int> biggestLosers = GenerateBigestLosersList();
+            if (biggestLosers.Count != players.Length)
+            {//if everyone doesnt win
+                foreach (int i in biggestLosers)
+                {
+                    players[i + winners.Count].GetComponent<Animator>().SetBool("Fourth", true); //set player crying animation
+                }
+            }
         }
+    }
+    #endregion
 
-        //foreach(RawImage image in images)
-        //{
-        //    image.gameObject.SetActive(false);
-        //}
-        for(int i = players.Length; i < podiums.Length; i++)
+    #region locational logic
+    void Display()
+    {
+        //moves players away from eachother to prevent accidental bumping
+        PrepPlayersForPlacement();
+
+        PlacePlayersAccordingly();
+    }
+
+    private void PlacePlayersAccordingly()
+    {
+        int incrementation = 0;
+        foreach (Player player in players)
         {
-            podiums[i].gameObject.SetActive(false);
+            //move to space above designated podium
+            player.transform.position = podiums[incrementation].position + Vector3.up * 30; //creates a 3 second fall for dramatic effect
+            player.transform.rotation = quickRot.transform.rotation;
+
+            //prep for next incrementation
+            incrementation++;
         }
-        Display(players);
-        StartCoroutine("Flash");
     }
 
+    private int PrepPlayersForPlacement()
+    {
+        int incrementation = 1;
+        foreach (Player player in players)
+        {
+            //prevents movement
+            Rigidbody rb = player.GetComponent<Rigidbody>();
+            rb.useGravity = false;
+            rb.velocity = Vector3.zero;
+            rb.constraints = RigidbodyConstraints.FreezeRotation | RigidbodyConstraints.FreezePositionX | RigidbodyConstraints.FreezePositionZ;
 
+            //moves to isolated location
+            player.transform.position = new Vector3(incrementation * 100, incrementation * 100, incrementation * 100);
 
+            //setup for next incrementation
+            incrementation++;
+        }
 
+        return incrementation;
+    }
 
-    void Display(Player[] players)
+    IEnumerator Reveal()
     {
-        int incrementation = 0;
+        //properly organize players to drop at same or different times
+        List<List<Player>> order = new List<List<Player>>();
```

Comment:
this seems overly complex for ordering players. Why nested list?

Reply:
Completed